File: android-app/app/src/main/java/com/example/shoumyo/ruinvolved/StudentActivity.java
package com.example.shoumyo.ruinvolved;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.MenuItem;
import android.view.View;
import android.widget.FrameLayout;

import com.example.shoumyo.ruinvolved.R;
import com.example.shoumyo.ruinvolved.data_sources.ClubsDataSource;
import com.example.shoumyo.ruinvolved.ui.MultiSelectionSpinner;
import com.example.shoumyo.ruinvolved.ui.fragments.ClubListFragment;
import com.example.shoumyo.ruinvolved.ui.fragments.ClubSearchFragment;

import java.util.HashMap;
import java.util.List;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class StudentActivity extends AppCompatActivity {

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_student);
        setFragment(new ClubSearchFragment());

        BottomNavigationView bottomNavigationView = findViewById(R.id.navigation);
        bottomNavigationView.setOnNavigationItemSelectedListener(
                menuItem -> {
                    switch (menuItem.getItemId()) {
                        case R.id.navigation_explore:
                            setFragment(new ClubSearchFragment());
                            return true;
                        case R.id.navigation_favorites:
                            setFragment(new ClubListFragment());
                            return true;
                        case R.id.navigation_map:
                            return false;
                    }

                    return false;
                }
        );
    }

    private void setFragment(Fragment fragment) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragment_container, fragment)
                .commit();
    }
}

File: android-app/app/src/main/java/com/example/shoumyo/ruinvolved/StudentActivity.kt
package com.example.shoumyo.ruinvolved

import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_student.*
import com.example.shoumyo.ruinvolved.ui.MultiSelectionSpinner



class StudentActivity : AppCompatActivity(), MultiSelectionSpinner.OnMultipleItemsSelectedListener {
    override fun selectedIndices(indices: MutableList<Int>?) {
    }

    override fun selectedStrings(strings: MutableList<String>?) {
    }

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.navigation_home -> {
//                message.setText(R.string.title_home)
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_dashboard -> {
//                message.setText(R.string.title_dashboard)
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_notifications -> {
//                message.setText(R.string.title_notifications)
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_student)

        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)

        val array = arrayOf("one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten")
        categories.setItems(array)
        categories.setSelection(intArrayOf(2, 6))
        categories.setListener(this)
    }
}
